pkg/model/config/config.go

```diff
@@ -209,16 +217,41 @@ func (c Config) Marshal() ([]byte, error) {
 	return content, nil
 }
 
-// Unmarshal unmarshals the bytes of a Config into c.
+// Unmarshal unmarshalls the bytes of a Config into c.
 func (c *Config) Unmarshal(b []byte) error {
 	if err := yaml.UnmarshalStrict(b, c); err != nil {
 		return fmt.Errorf("error unmarshalling project configuration: %v", err)
 	}
 
-	// Project versions < v3 do not support a plugins field.
-	if !c.IsV3() {
+	for i, r := range c.Resources {
+		// Restore regular plural that were omitted.
+		if r.Plural == "" {
+			c.Resources[i].Plural = resource.RegularPlural(r.Kind)
+		}
+		// Restore the default location that was omitted.
+		if r.Path == "" {
+			c.Resources[i].Path = resource.LocalPath(c.Repo, r.Group, r.Version, c.MultiGroup)
+		}
+		// Create a pointer to an empty struct instead of nil to prevent panics.
+		if r.API == nil {
+			c.Resources[i].API = &resource.API{}
+		}
+		// Create a pointer to an empty struct instead of nil to prevent panics.
+		if r.Webhooks == nil {
+			c.Resources[i].Webhooks = &resource.Webhooks{}
+		}
+	}
+
+	// Ignore some fields in v2.
+	if c.IsV2() {
+		// Only the default domain is allowed.
+		for i := range c.Resources {
+			c.Resources[i].Domain = c.Domain
+		}
+		// It doesn't support plugins.
 		c.Plugins = nil
 	}
+
```

Comment:
`Unmarshal` needs to be modified also because now the field `Resource.Domain` is part of its unique key, and also to set back the fields that we omitted in `Marshal`.